iquip/apps/logger.py

```diff
@@ -136,41 +147,74 @@ def __init__(self, name: str, parent: Optional[QObject] = None):
+        self.loggerFrame.fileLevelBox.textActivated.connect(self.setFileLevel)
+        self.loggerFrame.fileLevelBox.setCurrentText(self.levels_dict[self.fileHandler.level])
+
+    def initLogger(self):
+        """Initializes the root logger and handlers for constructor."""
+        self.levels_dict = {10: "DEBUG", 20: "INFO", 30: "WARNING", 40: "ERROR", 50: "CRITICAL"}
+        # initialize handlers
+        self.frameHandler = LoggingHandler(self.addLog)
+        self.fileHandler = handlers.TimedRotatingFileHandler(filename = "log", when = "midnight",
+                                                             interval = 1, encoding = 'utf-8')
+        self.fileHandler.suffix = '-%Y%m%d'
+        simpleFormat = "[%(name)s] %(message)s"
+        complexFormat = "%(asctime)s %(levelname)s [%(name)s]"\
+                        " [%(filename)s:%(lineno)d] %(message)s"
+        self.frameHandler.setFormatter(logging.Formatter(simpleFormat))
+        self.fileHandler.setFormatter(logging.Formatter(complexFormat))
+        # set rootLogger
         rootLogger = logging.getLogger()
-        rootLogger.addHandler(self.handler)
-        self.setLevel("WARNING")
-        self.loggerFrame.levelBox.addItems(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"])
-        self.loggerFrame.levelBox.textActivated.connect(self.setLevel)
-        self.loggerFrame.levelBox.setCurrentText("WARNING")
+        rootLogger.addHandler(self.frameHandler)
+        rootLogger.addHandler(self.fileHandler)
+        self.fileHandler.setLevel("WARNING")
+        self.setFrameLevel("WARNING")
+        self.setFileLevel("WARNING")
+
+
+    def frames(self) -> Tuple[LoggerFrame]:
+        """Overridden."""
+        return (self.loggerFrame,)
 
     @pyqtSlot(str)
-    def setLevel(self, levelText: str):
-        """Responds to the loggerFrame's levelBox widget and changes the handler's level.
+    def setFrameLevel(self, levelText: str):
+        """Responds to the loggerFrame's frameLevelBox widget and changes the handler's level.
 
         Args:
             leveltext: Selected level in the level select box.
               It should be one of "DEBUG", "INFO", "WARNING", "ERROR" and "CRITICAL".
               It should be case-sensitive and any other input is ignored.
         """
-        levels = {
-            "DEBUG": logging.DEBUG,
-            "INFO": logging.INFO,
-            "WARNING": logging.WARNING,
-            "ERROR": logging.ERROR,
-            "CRITICAL": logging.CRITICAL
-        }
-        if levelText in levels:
-            self.handler.setLevel(levels[levelText])
-            logging.getLogger().setLevel(levels[levelText])
+        if levelText in self.levels_dict.values():
+            self.frameHandler.setLevel(levelText)
+            if self.frameHandler.level < self.fileHandler.level:
+                logging.getLogger().setLevel(levelText)
+            else:
+                logging.getLogger().setLevel(self.levels_dict[self.fileHandler.level])
```

Comment:
How about:
```python
lowerLevel = min(self.frameHandler.level, self.fileHandler.level)
logging.getLogger().setLevel(self.levels_dict[lowerLevel])
```

Reply:
If you do this, `setFrameLevel()` and `setFileLevel()` will become almost identical, except for the handler (`frameHandler` and `fileHandler`, respectively).
Then you can merge them as (for example):
```python
def setLevel(self, handler_: logging.Handler, levelText: str):
    ...
```
and then use [`functools.partial()`](https://docs.python.org/3/library/functools.html#functools.partial) when you connect them as slots.